### DISCUSSION TITLE
curl_easygetinfo with CURLINFO_LOCAL_PORT option returns -1 in port.

### DISCUSSION THREAD

**Discussion by bansalnehal** [curl/curl#14659] [Q&A]
Hi, I am generating a set of 100 HTTP GET requests per second to an HTTP server. I have set a timeout of 3 seconds for each request using CURLOPT_TIMEOUT. Out of 100, 1 or 2 requests return an error code of 28 in msg returned from curl_multi_info_read. Following are some values which I extracted from this msg: 
1) CURLINFO_TOTAL_TIME - 3.0012,
2) CURLINFO_CONNECT_TIME - 0.00050,
3) CURLINFO_EFFECTIVE_URL - A valid GET request is printed here,
4) CURLINFO_STARTTRANSFER_TIME - 0.00000,
5) CURLINFO_LOCAL_PORT - (-1),

What should I infer from these values? Did the request reach the server or did it time out with the multi handle itself? 

Curl - 7.76

**Comment by bagder** [comment]
> Did the request reach the server

I believe it did not do a successful TCP connect handshake.

**Reply by bansalnehal** [reply]
Where could the fault be, at server or client for the failure of handshake?

**Reply by bansalnehal** [reply]
Is there a parameter that can tell that the TCP handshake has failed?
